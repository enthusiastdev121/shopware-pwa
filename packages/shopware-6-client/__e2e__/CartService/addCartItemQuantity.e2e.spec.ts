import {
  update,
  addCartItemQuantity,
  addProductToCart
} from "@shopware-pwa/shopware-6-client";

describe("shopware-6-client - E2E - CartService - addCartItemQuantity", () => {
  beforeEach(() => {
    update({ contextToken: "" });
  });

  it("should add cart item quantity response", async () => {
    await addProductToCart("9cce06f9dc424844989a06cfe3dc98da", 2);
    const result = await addCartItemQuantity(
      "9cce06f9dc424844989a06cfe3dc98da",
      2
    );
    result.token = "mockedToken";
    result.deliveries = result.deliveries.map((delivery: any) => {
      delivery.deliveryDate.earliest = "mockedDate";
      delivery.deliveryDate.latest = "mockedDate";
      delivery.positions = delivery.positions.map((position: any) => {
        position.deliveryDate.earliest = "mockedDate";
        position.deliveryDate.latest = "mockedDate";
        return position;
      });
      return delivery;
    });
    expect(result).toMatchSnapshot();
  });

  it("should test add cart item quantity error response", async () => {
    try {
      await addCartItemQuantity("044a190a54ab4f06803909c3ee8063ef", 2);
      expect("didn't throw an error").toEqual("should throw an error");
    } catch (e) {
      expect(e).toMatchSnapshot();
    }
  });
});
