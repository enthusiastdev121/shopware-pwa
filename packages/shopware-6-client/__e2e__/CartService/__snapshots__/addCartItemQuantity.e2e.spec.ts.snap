// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopware-6-client - E2E - CartService - addCartItemQuantity should add cart item quantity response 1`] = `
Object {
  "deliveries": Array [
    Object {
      "deliveryDate": Object {
        "earliest": "mockedValue",
        "extensions": Array [],
        "latest": "mockedValue",
      },
      "extensions": Array [],
      "location": Object {
        "address": null,
        "country": Object {
          "_uniqueIdentifier": "3ff87c8746dc40ecaea1495eb254278e",
          "active": true,
          "createdAt": "2019-10-08T09:41:13+00:00",
          "customFields": null,
          "customerAddresses": null,
          "displayStateInRegistration": false,
          "extensions": Object {
            "foreignKeys": Object {
              "_uniqueIdentifier": null,
              "createdAt": null,
              "extensions": Array [],
              "translated": Array [],
              "updatedAt": null,
              "versionId": null,
            },
          },
          "forceStateInRegistration": false,
          "id": "3ff87c8746dc40ecaea1495eb254278e",
          "iso": "DE",
          "iso3": "DEU",
          "name": "Germany",
          "orderAddresses": null,
          "position": 1,
          "salesChannelDefaultAssignments": null,
          "salesChannels": null,
          "shippingAvailable": true,
          "states": null,
          "taxFree": false,
          "taxRules": null,
          "translated": Object {
            "customFields": Array [],
            "name": "Germany",
          },
          "translations": null,
          "updatedAt": null,
          "versionId": null,
        },
        "extensions": Array [],
        "state": null,
      },
      "positions": Array [
        Object {
          "deliveryDate": Object {
            "earliest": "mockedValue",
            "extensions": Array [],
            "latest": "mockedValue",
          },
          "extensions": Array [],
          "identifier": "9cce06f9dc424844989a06cfe3dc98da",
          "lineItem": Object {
            "children": Array [],
            "cover": Object {
              "_uniqueIdentifier": "0eff54b507d3405ab9cd168e7247fce1",
              "alt": null,
              "avatarUser": null,
              "categories": null,
              "cmsBlocks": null,
              "cmsPages": null,
              "cmsSections": null,
              "createdAt": "2019-12-05T14:00:28+00:00",
              "customFields": null,
              "documentBaseConfigs": null,
              "documents": null,
              "extensions": Object {
                "foreignKeys": Object {
                  "_uniqueIdentifier": null,
                  "createdAt": null,
                  "extensions": Array [],
                  "translated": Array [],
                  "updatedAt": null,
                  "versionId": null,
                },
                "internal_mapping_storage": Object {
                  "_uniqueIdentifier": null,
                  "createdAt": null,
                  "extensions": Array [],
                  "translated": Array [],
                  "updatedAt": null,
                  "versionId": null,
                },
              },
              "fileExtension": "png",
              "fileName": "fashion_ZA028248_007_p_5.png",
              "fileSize": 303299,
              "hasFile": true,
              "id": "0eff54b507d3405ab9cd168e7247fce1",
              "mailTemplateMedia": null,
              "mediaFolder": null,
              "mediaFolderId": null,
              "mediaType": Object {
                "extensions": Array [],
                "flags": Array [
                  "transparent",
                ],
                "name": "IMAGE",
              },
              "metaData": Object {
                "height": 906,
                "type": 3,
                "width": 600,
              },
              "mimeType": "image/png",
              "orderLineItems": null,
              "paymentMethods": null,
              "private": false,
              "productConfiguratorSettings": null,
              "productManufacturers": null,
              "productMedia": null,
              "propertyGroupOptions": null,
              "shippingMethods": null,
              "tags": null,
              "thumbnails": Array [],
              "thumbnailsRo": "mockedValue",
              "title": null,
              "translated": Object {
                "alt": null,
                "customFields": Array [],
                "title": null,
              },
              "translations": null,
              "updatedAt": "2019-12-05T14:00:29+00:00",
              "uploadedAt": "2019-12-05T14:00:29+00:00",
              "url": "https://shopware6-demo.vuestorefront.io/media/af/1e/04/1575554429/fashion_ZA028248_007_p_5.png.png",
              "user": null,
              "userId": "e7fcae1311294d9ea3b8973909529b5f",
              "versionId": null,
            },
            "deliveryInformation": Object {
              "deliveryTime": null,
              "extensions": Array [],
              "freeDelivery": false,
              "restockTime": 3,
              "stock": 36,
              "weight": 0,
            },
            "description": null,
            "extensions": Array [],
            "good": true,
            "id": "9cce06f9dc424844989a06cfe3dc98da",
            "label": "Aerodynamic Iron Exchangesound",
            "modified": true,
            "payload": Object {
              "categories": Array [
                "3f637f17cd9f4891a2d7625d19fb37c9",
                "83e555dee2d2425eba42b7c628b65c67",
                "31069346fab64ee591343728e09bf913",
              ],
              "options": Array [],
              "productNumber": "6658ff91b2de47948b3787cc0a43c656",
              "properties": Array [
                "20beab74a1284d05b9661f973bbc6a4b",
                "282369ca18f441168d34dfd853af01de",
                "2e0fd2231732496caa810a16fc1b7f51",
                "2fab6eebc75e43899b833f183adc1939",
                "3b445efcc4a34c47bea88b55fc215a07",
                "44353d5f4b2f460fa9015ee523243bb6",
                "52593112fd4a4630a39299dd318f0fb7",
                "5ce324943a694a1d833dcc6b798f1a47",
                "5ffb3e52edc244549efd1331fb1b60ac",
                "6a29887639614dc59380245a3baaf1b6",
                "7f2802a334504e0b921295862da6087d",
                "8414633917e64d16ada513a5f170ed41",
                "872a36ad229f4a2bb747e8f504bd8d67",
                "a4896fec5a3e496f8f8d5f50e91cc09d",
                "b27071703d9c4bbe902bdf8d593035d3",
                "c86ec38d46674fb593c3f1ffc3b1a523",
                "e908d80a5e104c50b78cc1b170f331b2",
                "eea312ba945240289629e00bb9e8a90f",
              ],
              "tags": null,
            },
            "price": Object {
              "calculatedTaxes": Array [
                Object {
                  "extensions": Array [],
                  "price": 40,
                  "tax": 6.39,
                  "taxRate": 19,
                },
              ],
              "extensions": Array [],
              "quantity": 4,
              "referencePrice": null,
              "taxRules": Array [
                Object {
                  "extensions": Array [],
                  "percentage": 100,
                  "taxRate": 19,
                },
              ],
              "totalPrice": 40,
              "unitPrice": 10,
            },
            "priceDefinition": Object {
              "extensions": Array [],
              "isCalculated": true,
              "precision": 2,
              "price": 10,
              "quantity": 4,
              "referencePriceDefinition": null,
              "taxRules": Array [
                Object {
                  "extensions": Array [],
                  "percentage": 100,
                  "taxRate": 19,
                },
              ],
              "type": "quantity",
            },
            "quantity": 4,
            "quantityInformation": Object {
              "extensions": Array [],
              "maxPurchase": null,
              "minPurchase": 1,
              "purchaseSteps": 1,
            },
            "referencedId": "9cce06f9dc424844989a06cfe3dc98da",
            "removable": true,
            "requirement": null,
            "stackable": true,
            "type": "product",
          },
          "price": Object {
            "calculatedTaxes": Array [
              Object {
                "extensions": Array [],
                "price": 40,
                "tax": 6.39,
                "taxRate": 19,
              },
            ],
            "extensions": Array [],
            "quantity": 4,
            "referencePrice": null,
            "taxRules": Array [
              Object {
                "extensions": Array [],
                "percentage": 100,
                "taxRate": 19,
              },
            ],
            "totalPrice": 40,
            "unitPrice": 10,
          },
          "quantity": 4,
        },
      ],
      "shippingCosts": Object {
        "calculatedTaxes": Array [
          Object {
            "extensions": Array [],
            "price": 0,
            "tax": 0,
            "taxRate": 19,
          },
        ],
        "extensions": Array [],
        "quantity": 1,
        "referencePrice": null,
        "taxRules": Array [
          Object {
            "extensions": Array [],
            "percentage": 100,
            "taxRate": 19,
          },
        ],
        "totalPrice": 0,
        "unitPrice": 0,
      },
      "shippingMethod": Object {
        "_uniqueIdentifier": "45dd857987ba49b28c63c5892e8aa0f5",
        "active": true,
        "availabilityRule": null,
        "availabilityRuleId": "f15832556ab647928ac0525a01f3c3ac",
        "createdAt": "2019-10-08T09:41:16+00:00",
        "customFields": null,
        "deliveryTime": Object {
          "_uniqueIdentifier": "7b109f96ff984fae912caca48bf07e9e",
          "createdAt": "2019-10-08T09:41:16+00:00",
          "customFields": null,
          "extensions": Object {
            "foreignKeys": Object {
              "_uniqueIdentifier": null,
              "createdAt": null,
              "extensions": Array [],
              "translated": Array [],
              "updatedAt": null,
              "versionId": null,
            },
            "internal_mapping_storage": Object {
              "_uniqueIdentifier": null,
              "createdAt": null,
              "extensions": Array [],
              "translated": Array [],
              "updatedAt": null,
              "versionId": null,
            },
          },
          "id": "7b109f96ff984fae912caca48bf07e9e",
          "max": 3,
          "min": 1,
          "name": "1-3 days",
          "products": null,
          "shippingMethods": null,
          "translated": Object {
            "customFields": Array [],
            "name": "1-3 days",
          },
          "translations": null,
          "unit": "day",
          "updatedAt": null,
          "versionId": null,
        },
        "deliveryTimeId": "7b109f96ff984fae912caca48bf07e9e",
        "description": null,
        "extensions": Object {
          "foreignKeys": Object {
            "_uniqueIdentifier": null,
            "createdAt": null,
            "extensions": Array [],
            "translated": Array [],
            "updatedAt": null,
            "versionId": null,
          },
        },
        "id": "45dd857987ba49b28c63c5892e8aa0f5",
        "media": null,
        "mediaId": null,
        "name": "Standard",
        "orderDeliveries": null,
        "prices": Array [
          Object {
            "_uniqueIdentifier": "3db7e537f97145cebccc6b036ae0a292",
            "calculation": 1,
            "calculationRule": null,
            "calculationRuleId": null,
            "createdAt": "2019-10-08T09:41:16+00:00",
            "currency": null,
            "currencyId": "b7d2554b0ce847cd82f3ac9bd1c0dfca",
            "customFields": null,
            "extensions": Object {
              "foreignKeys": Object {
                "_uniqueIdentifier": null,
                "createdAt": null,
                "extensions": Array [],
                "translated": Array [],
                "updatedAt": null,
                "versionId": null,
              },
            },
            "id": "3db7e537f97145cebccc6b036ae0a292",
            "price": 0,
            "quantityEnd": null,
            "quantityStart": 0,
            "rule": null,
            "ruleId": null,
            "shippingMethod": null,
            "shippingMethodId": "45dd857987ba49b28c63c5892e8aa0f5",
            "translated": Array [],
            "updatedAt": null,
            "versionId": null,
          },
        ],
        "salesChannelDefaultAssignments": null,
        "salesChannels": null,
        "tags": null,
        "translated": Object {
          "customFields": Array [],
          "description": null,
          "name": "Standard",
        },
        "translations": null,
        "updatedAt": null,
        "versionId": null,
      },
    },
  ],
  "errors": Array [],
  "extensions": Array [],
  "lineItems": Array [
    Object {
      "children": Array [],
      "cover": Object {
        "_uniqueIdentifier": "0eff54b507d3405ab9cd168e7247fce1",
        "alt": null,
        "avatarUser": null,
        "categories": null,
        "cmsBlocks": null,
        "cmsPages": null,
        "cmsSections": null,
        "createdAt": "2019-12-05T14:00:28+00:00",
        "customFields": null,
        "documentBaseConfigs": null,
        "documents": null,
        "extensions": Object {
          "foreignKeys": Object {
            "_uniqueIdentifier": null,
            "createdAt": null,
            "extensions": Array [],
            "translated": Array [],
            "updatedAt": null,
            "versionId": null,
          },
          "internal_mapping_storage": Object {
            "_uniqueIdentifier": null,
            "createdAt": null,
            "extensions": Array [],
            "translated": Array [],
            "updatedAt": null,
            "versionId": null,
          },
        },
        "fileExtension": "png",
        "fileName": "fashion_ZA028248_007_p_5.png",
        "fileSize": 303299,
        "hasFile": true,
        "id": "0eff54b507d3405ab9cd168e7247fce1",
        "mailTemplateMedia": null,
        "mediaFolder": null,
        "mediaFolderId": null,
        "mediaType": Object {
          "extensions": Array [],
          "flags": Array [
            "transparent",
          ],
          "name": "IMAGE",
        },
        "metaData": Object {
          "height": 906,
          "type": 3,
          "width": 600,
        },
        "mimeType": "image/png",
        "orderLineItems": null,
        "paymentMethods": null,
        "private": false,
        "productConfiguratorSettings": null,
        "productManufacturers": null,
        "productMedia": null,
        "propertyGroupOptions": null,
        "shippingMethods": null,
        "tags": null,
        "thumbnails": Array [],
        "thumbnailsRo": "mockedValue",
        "title": null,
        "translated": Object {
          "alt": null,
          "customFields": Array [],
          "title": null,
        },
        "translations": null,
        "updatedAt": "2019-12-05T14:00:29+00:00",
        "uploadedAt": "2019-12-05T14:00:29+00:00",
        "url": "https://shopware6-demo.vuestorefront.io/media/af/1e/04/1575554429/fashion_ZA028248_007_p_5.png.png",
        "user": null,
        "userId": "e7fcae1311294d9ea3b8973909529b5f",
        "versionId": null,
      },
      "deliveryInformation": Object {
        "deliveryTime": null,
        "extensions": Array [],
        "freeDelivery": false,
        "restockTime": 3,
        "stock": 36,
        "weight": 0,
      },
      "description": null,
      "extensions": Array [],
      "good": true,
      "id": "9cce06f9dc424844989a06cfe3dc98da",
      "label": "Aerodynamic Iron Exchangesound",
      "modified": false,
      "payload": Object {
        "categories": Array [
          "3f637f17cd9f4891a2d7625d19fb37c9",
          "83e555dee2d2425eba42b7c628b65c67",
          "31069346fab64ee591343728e09bf913",
        ],
        "options": Array [],
        "productNumber": "6658ff91b2de47948b3787cc0a43c656",
        "properties": Array [
          "20beab74a1284d05b9661f973bbc6a4b",
          "282369ca18f441168d34dfd853af01de",
          "2e0fd2231732496caa810a16fc1b7f51",
          "2fab6eebc75e43899b833f183adc1939",
          "3b445efcc4a34c47bea88b55fc215a07",
          "44353d5f4b2f460fa9015ee523243bb6",
          "52593112fd4a4630a39299dd318f0fb7",
          "5ce324943a694a1d833dcc6b798f1a47",
          "5ffb3e52edc244549efd1331fb1b60ac",
          "6a29887639614dc59380245a3baaf1b6",
          "7f2802a334504e0b921295862da6087d",
          "8414633917e64d16ada513a5f170ed41",
          "872a36ad229f4a2bb747e8f504bd8d67",
          "a4896fec5a3e496f8f8d5f50e91cc09d",
          "b27071703d9c4bbe902bdf8d593035d3",
          "c86ec38d46674fb593c3f1ffc3b1a523",
          "e908d80a5e104c50b78cc1b170f331b2",
          "eea312ba945240289629e00bb9e8a90f",
        ],
        "tags": null,
      },
      "price": Object {
        "calculatedTaxes": Array [
          Object {
            "extensions": Array [],
            "price": 40,
            "tax": 6.39,
            "taxRate": 19,
          },
        ],
        "extensions": Array [],
        "quantity": 4,
        "referencePrice": null,
        "taxRules": Array [
          Object {
            "extensions": Array [],
            "percentage": 100,
            "taxRate": 19,
          },
        ],
        "totalPrice": 40,
        "unitPrice": 10,
      },
      "priceDefinition": Object {
        "extensions": Array [],
        "isCalculated": true,
        "precision": 2,
        "price": 10,
        "quantity": 4,
        "referencePriceDefinition": null,
        "taxRules": Array [
          Object {
            "extensions": Array [],
            "percentage": 100,
            "taxRate": 19,
          },
        ],
        "type": "quantity",
      },
      "quantity": 4,
      "quantityInformation": Object {
        "extensions": Array [],
        "maxPurchase": null,
        "minPurchase": 1,
        "purchaseSteps": 1,
      },
      "referencedId": "9cce06f9dc424844989a06cfe3dc98da",
      "removable": true,
      "requirement": null,
      "stackable": true,
      "type": "product",
    },
  ],
  "modified": false,
  "name": "8a243080f92e4c719546314b577cf82b",
  "price": Object {
    "calculatedTaxes": Array [
      Object {
        "extensions": Array [],
        "price": 40,
        "tax": 6.39,
        "taxRate": 19,
      },
    ],
    "extensions": Array [],
    "netPrice": 33.61,
    "positionPrice": 40,
    "taxRules": Array [
      Object {
        "extensions": Array [],
        "percentage": 100,
        "taxRate": 19,
      },
    ],
    "taxStatus": "gross",
    "totalPrice": 40,
  },
  "token": "mockedValue",
  "transactions": Array [
    Object {
      "amount": Object {
        "calculatedTaxes": Array [
          Object {
            "extensions": Array [],
            "price": 40,
            "tax": 6.39,
            "taxRate": 19,
          },
        ],
        "extensions": Array [],
        "quantity": 1,
        "referencePrice": null,
        "taxRules": Array [
          Object {
            "extensions": Array [],
            "percentage": 100,
            "taxRate": 19,
          },
        ],
        "totalPrice": 40,
        "unitPrice": 40,
      },
      "extensions": Array [],
      "paymentMethodId": "cad2356ee68c473ca7da517d0ad32de1",
    },
  ],
}
`;

exports[`shopware-6-client - E2E - CartService - addCartItemQuantity should test add cart item quantity error response 1`] = `[Error: Request failed with status code 500]`;
